### PR TITLE
addition of py_ast

### PR DESCRIPTION
**Pull request by reshinthadithyan**
The dataset consists of parsed Parsed ASTs that were used to train and evaluate the DeepSyn tool. 
The Python programs are collected from GitHub repositories
by removing duplicate files, removing project forks (copy of another existing repository)
,keeping only programs that parse and have at most 30'000 nodes in the AST and 
we aim to remove obfuscated files

### CODE REVIEW

File: datasets/py_ast/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+- [Additional Information](#additional-information)
+  - [Dataset Curators](#dataset-curators)
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
+
+## Dataset Description
+
+- **[py150](https://www.sri.inf.ethz.ch/py150)** 
+- **py150**
+- **[Probabilistic Model for Code with Decision Trees](https://dl.acm.org/doi/10.1145/3022671.2984041)**
```
Comment (yjernite) [inline datasets/py_ast/README.md:35]:
```suggestion
- **homepage**: [py150](https://www.sri.inf.ethz.ch/py150) 
- **Paper**: [Probabilistic Model for Code with Decision Trees](https://dl.acm.org/doi/10.1145/3022671.2984041)
```

**Comment by yjernite** [comment]
Hi @reshinthadithyan !

As mentioned on the Slack, it would be better in this case to parse the file lines into the following feature structure:
```python
"ast": datasets.Sequence(
    {
        "type": datasets.Value("string"),
        "value": datasets.Value("string"),
        "children": datasets.Sequence(datasets.Value("int32")),
    },
)
```

Here are a few more things to fix before we can move forward:
- the class name needs to be the CamelCase equivalent of the script name, so here it will have to be `PyAst`
- the `README.md` needs to have the tags at the top
- The homepage/info list at the top should be in the same format as the template (added a suggestion)
- You should add the dataset tags and field description to the README as described here: https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

Good luck, let us know if you need any help!

**Comment by reshinthadithyan** [comment]
Hello @yjernite, changes are made as we talked. Hope this would suffice. Thanks. Feel free to point out any room to improvement.

**Comment by yjernite** [comment]
Good progress! Here's what still needs to be done:
- first, you need to rebase to master for the tests to pass :)
- the information in your `Data Fields` paragraph should go into `Data Instances`. Data fields should describe the fields one by one, as in e.g. https://github.com/huggingface/datasets/tree/master/datasets/eli5#data-fields
- you still need to add the YAML tags obtained with the tagging app

Should be good to go after that!

**Comment by reshinthadithyan** [comment]
Hello @yjernite, changes as talked are being done.

**Comment by lhoestq** [comment]
Looks like this PR includes changes about many other files than the ones for py_ast

Could you create another branch and another PR please ?
